### ISSUE TITLE
Compatibility 

### ISSUE THREAD

**Issue by JLundberg65** [Tannz0rz/Mandoline#1]
Hello,
I really like the feasibility of your slicing routine as a hatch fill.  I have a project I'm playing with that's in Visual Studio C and C++.  Since I'm not using Ubuntu and its dependencies, is it possible to make a version that would compile in Visual Studio?

John.

**Comment by Tannz0rz** [comment]
It ought to sans Cairo and the main.cpp program provided; Mandoline itself should compile so long as you can manage to link your program to Eigen, which can be found here: https://eigen.tuxfamily.org/index.php?title=Main_Page

Note that I have not tried to build on Windows, but I am certain it should.

**Comment by JLundberg65** [comment]
Casey,

 

I appreciate your reply.  Thank you.

 

Just a couple of questions if you wouldn’t mind.

 

How fast would you say your process is?  Let’s say this was passed 30 small objects about .75” height and .25” width  over a 10” x 10” page and the line spacing was about .010”

How difficult would it be to set a main hatch angle of say, 20deg, 45deg or 90deg?

Of course all id mute if I can’t get it and Eigen to compile in Visual Studio 2013 

 

Thanks again,

 

John

**Comment by Tannz0rz** [comment]
By "height" and "width" are you referring to two dimensions? This is dependent upon the complexity of the objects, increasing calculation time given the number of total sides to be compared against. When I get the chance I'd be happy to create & slice 30 20-sided polygons embedded in a rectangle at those specifications and report to you my findings.

**Comment by JLundberg65** [comment]
Correct, 2 dimensional objects. Will it also work with Bezier curves as well?  Here are some objects in my thought process.  Others could be objects like your example.  How long did that take to process?  Not actually draw but process the job in memory?

The arcs could be Bezier or made from vectors.  The time it takes to fill the inside of 30 of these be great. If a box was put around the extents of the word and the box was filled leaving the text hollow would be another type of object.  I appreciate you offering to run a test.  If the calculations of a single object, “T” takes a while when multiplied out, maybe start up different threads to handle multiple objects simultaneously.  Just rambling J 

 

The objects XY Dim are say .250, .750

 



Thanks again.  

 

Maybe we could email privately?

 

Take care and stay safe.

.

**Comment by Tannz0rz** [comment]
Certainly, please contact me at casey.sanchez.94@gmail.com

https://i.imgur.com/Kk5hQ7F.png

This took 1283ms, which certainly is not the most performant. It could stand to be optimized in various aspects I'm sure.

Mandoline works by utilizing the even-odd rule (https://en.wikipedia.org/wiki/Even%E2%80%93odd_rule), which requires a set of line segments. It may be possible to approximate Bezier curves using line segments to your desired precision to achieve a similar effect.
